bridge/client/app/_components/ktb-http-loading-bar/ktb-http-loading-bar.component.ts

```diff
@@ -1,13 +1,17 @@
-import {Component, Input, OnInit} from '@angular/core';
+import {Component, Input, OnDestroy, OnInit} from '@angular/core';
 import {HttpStateService} from "../../_services/http-state.service";
 import {HttpProgressState, HttpState} from "../../_models/http-progress-state";
+import {Subject} from "rxjs";
+import {takeUntil} from "rxjs/operators";
 
 @Component({
   selector: 'ktb-http-loading-bar',
   templateUrl: './ktb-http-loading-bar.component.html',
   styleUrls: ['./ktb-http-loading-bar.component.scss']
 })
-export class KtbHttpLoadingBarComponent implements OnInit {
+export class KtbHttpLoadingBarComponent implements OnInit, OnDestroy {
```

Comment:
This class and KtbHttpLoadingSpinnerComponent have the loading flag, the show and hide funcitons and the subscribe to the httpState in common, may be extracting this parts to a general HttpLoadingState component could be an option to reduce duplicates.

Reply:
Good point.
I actually removed the `KtbHttpLoadingSpinnerComponent` since it was never used. And actually I would implement content projection to `KtbHttpLoadingBarComponent` and then the developer could choose to either add a loading spinner, loading bar or show any other component while something is loading.